app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+package com.rockthevote.grommet.ui.registration
+
+import android.util.Base64
+import com.rockthevote.grommet.R
+import com.rockthevote.grommet.data.db.model.*
+import com.rockthevote.grommet.ui.registration.address.AddressData
+import com.rockthevote.grommet.ui.registration.name.PersonNameData
+import com.rockthevote.grommet.util.Dates
+import java.util.*
+
+/**
+ * Handles transformation of [registrationData]
+ */
+class RegistrationDataTransformer @Throws(InvalidRegistrationException::class) constructor(
+    private val registrationData: RegistrationData,
+    private val partnerInformation: PartnerInformation
+) {
+
+    init {
+        validateRegistrationData(registrationData)
+    }
+
+    // These have already be evaluated as non null in [validateRegistrationData]
+    private val newRegistrationData = registrationData.newRegistrantData!!
+    private val addressData = registrationData.addressData!!
+    private val additionalInfoData = registrationData.additionalInfoData!!
+    private val assistanceData = registrationData.assistanceData!!
+    private val reviewData = registrationData.reviewData!!
+
+    /**
+     * Transforms the [registrationData] and [partnerInformation] into a [RockyRequest]
+     *
+     * @throws [InvalidRegistrationException] if unable to validate [RegistrationData]
+     */
+    fun transform(
+        unknown: UnknownDataSource
+    ): RockyRequest {
+
+        val body = RockyRequestBody(
+            lang = reviewData.formLanguage,
+            phoneType = null, // TODO I can't find where/what is supposed to populate this
+            partnerId = partnerInformation.partnerId,
+            optInEmail = additionalInfoData.hasOptedIntoNewsUpdates,
+            optInSms = additionalInfoData.hasOptedIntoNewsCallAndText,
+            optInVolunteer = additionalInfoData.hasOptedForVolunteerText,
+            // TODO All of these data sources are unknown, I think they may be from partner information, but don't know for sure
```

Comment:
ugh, more grossness I'm afraid. 

tl/dr: they're not used anywhere, they have [defaults](https://github.com/TrustTheVote-Project/Grommet/blob/8c10e8b88a44b5e28c6161cf71685b75d519a05b/app/src/main/java/com/rockthevote/grommet/data/db/model/RockyRequest.java#L28) set in the original database. We have to submit them, just use the default values.

Long Version:
The API format we're using is an official one, and is (was?) supposed to be the new standard for all voter registration requests anywhere. Because of that, there are a lot of values we may or may not use.

The database was originally designed to capture all this information, but things happened (or didn't) and then they never got used.

Reply:
[here's](https://github.com/TrustTheVote-Project/Grommet/blob/6b0230af2e1f500db3943544cc64a177bd44d0ed/app/src/main/java/com/rockthevote/grommet/data/api/model/ApiRockyRequest.java#L68) another example of the default's being set in the builder and then "overidden" in the `fromApi` method with...the database default values.

Reply:
So is it safe to assume that anything preset in the Builder of the old ApiRequests is a default value that won't change?

Reply:
I see what most of them are now, I had to do a clean checkout of Master to hunt some of these down. Anything that had a hard value set that I could find, I set it to that. What I'm left with is:
```
            sourceTrackingId = session.sourceTrackingId,
            partnerTrackingId = session.partnerTrackingId,
            geoLocation = session.geoLocation,
            openTrackingId = session.openTrackingId,
```
Which appears to come from Session data. I've renamed `UnknownDataSource` to `SessionData`. This should be easy enough to derive and build from the session stored in the database as we go to submit this.